### ISSUE TITLE
Watcher reports L1[0] corruption on cq_prefetcher core on L chip

### ISSUE THREAD

**Issue by tt-dma** [tenstorrent-metal/tt-metal#6159]
Steps to reproduce, on n300

- `git checkout dma/4940_watcher_asserts`
- `make build && make tests`
- `./build/test/tt_metal/unit_tests_fast_dispatch --gtest_filter=*WatcherAssertBrisc`

Test should fail and if you look up in the log you'll see.
```
                 Always | FATAL    | Watcher found corruption at L1[0] on core (x=2,y=10): read 1
              LLRuntime | INFO     | Watcher thread stopped watching...
```

This test was passing before rebasing on latest fast dispatch changes, and we also see this issue on `*TestWatcherWaypoints` after enabling it to run on R chip (#6096).

**Comment by DrJessop** [comment]
Pushed fix to your branch: 53ca98daca8c578c6e73d81cdbee971fc7a17b1e

Turns out, we don't error out when we compile kernels with uninitialized variables. I made an issue here: https://github.com/tenstorrent-metal/tt-metal/issues/6250

**Comment by tt-dma** [comment]
Tested and merged to main as part of my PR: https://github.com/tenstorrent-metal/tt-metal/pull/6175

Thanks for the fix!
